### DISCUSSION TITLE
Trouble running Trilium server on RaspberryPi

### DISCUSSION THREAD

**Discussion by Korozion** [TriliumNext/Trilium#2136] [Q&A] [answered]
Hi !
I am trying to install the Trilium server on a raspberrypi 3B+ running OSMC, but I can't get it to work.

I followed the "Manual server installation" from the wiki.
When running the command `nohup node src/www &`, I get  `nohup: ignoring input and appending output to 'nohup.out'`
Then nothing happens until I press a key, which leads to `[1]+  Exit 1                  nohup node src/www`, and the Trilium server is not accessible via my browser ("unable to connect").

OSMC is up-to-date,
nodejs version is 12.22.5
npm version is 7.21.1
nohup version is 8.30
I had to run `npm audit fix` and `npm audit fix --force` to get rid of some vulnerabilities.
Tried running the nohup command as root and as standard user with same results.
Tried installing the alleged missing files with `npm install <missing-file>`
Tried removing whole "trilium" folder and start over, even after raspberrypi reboot.
None of the above helped.

I pasted the nohup.out below

What should I try ?
Thanks !

> /home/osmc/trilium/node_modules/bindings/bindings.js:135
>   throw err;
>   ^
> 
> Error: Could not locate the bindings file. Tried:
>   ^f^r /home/osmc/trilium/node_modules/better-sqlite3/build/better_sqlite3.node
>   ^f^r /home/osmc/trilium/node_modules/better-sqlite3/build/Debug/better_sqlite3.node
>   ^f^r /home/osmc/trilium/node_modules/better-sqlite3/build/Release/better_sqlite3.node
>   ^f^r /home/osmc/trilium/node_modules/better-sqlite3/out/Debug/better_sqlite3.node
>   ^f^r /home/osmc/trilium/node_modules/better-sqlite3/Debug/better_sqlite3.node
>   ^f^r /home/osmc/trilium/node_modules/better-sqlite3/out/Release/better_sqlite3.node
>   ^f^r /home/osmc/trilium/node_modules/better-sqlite3/Release/better_sqlite3.node
>   ^f^r /home/osmc/trilium/node_modules/better-sqlite3/build/default/better_sqlite3.node
>   ^f^r /home/osmc/trilium/node_modules/better-sqlite3/compiled/12.22.5/linux/arm/better_sqlite3.node
>   ^f^r /home/osmc/trilium/node_modules/better-sqlite3/addon-build/release/install-root/better_sqlite3.node
>   ^f^r /home/osmc/trilium/node_modules/better-sqlite3/addon-build/debug/install-root/better_sqlite3.node
>   ^f^r /home/osmc/trilium/node_modules/better-sqlite3/addon-build/default/install-root/better_sqlite3.node
>   ^f^r /home/osmc/trilium/node_modules/better-sqlite3/lib/binding/node-v72-linux-arm/better_sqlite3.node
>     at bindings (/home/osmc/trilium/node_modules/bindings/bindings.js:126:9)
>     at Object.<anonymous> (/home/osmc/trilium/node_modules/better-sqlite3/lib/database.js:9:24)
>     at Module._compile (internal/modules/cjs/loader.js:999:30)
>     at Object.Module._extensions..js (internal/modules/cjs/loader.js:1027:10)
>     at Module.load (internal/modules/cjs/loader.js:863:32)
>     at Function.Module._load (internal/modules/cjs/loader.js:708:14)
>     at Module.require (internal/modules/cjs/loader.js:887:19)
>     at require (internal/modules/cjs/helpers.js:74:18)
>     at Object.<anonymous> (/home/osmc/trilium/node_modules/better-sqlite3/lib/index.js:2:18)
>     at Module._compile (internal/modules/cjs/loader.js:999:30) {
>   tries: [
>     '/home/osmc/trilium/node_modules/better-sqlite3/build/better_sqlite3.node',
>     '/home/osmc/trilium/node_modules/better-sqlite3/build/Debug/better_sqlite3.node',
>     '/home/osmc/trilium/node_modules/better-sqlite3/build/Release/better_sqlite3.node',
>     '/home/osmc/trilium/node_modules/better-sqlite3/out/Debug/better_sqlite3.node',
>     '/home/osmc/trilium/node_modules/better-sqlite3/Debug/better_sqlite3.node',
>     '/home/osmc/trilium/node_modules/better-sqlite3/out/Release/better_sqlite3.node',
>     '/home/osmc/trilium/node_modules/better-sqlite3/Release/better_sqlite3.node',
>     '/home/osmc/trilium/node_modules/better-sqlite3/build/default/better_sqlite3.node',
>     '/home/osmc/trilium/node_modules/better-sqlite3/compiled/12.22.5/linux/arm/better_sqlite3.node',
>     '/home/osmc/trilium/node_modules/better-sqlite3/addon-build/release/install-root/better_sqlite3.node',
>    '/home/osmc/trilium/node_modules/better-sqlite3/addon-build/debug/install-root/better_sqlite3.node',
>    '/home/osmc/trilium/node_modules/better-sqlite3/addon-build/default/install-root/better_sqlite3.node',
>    '/home/osmc/trilium/node_modules/better-sqlite3/lib/binding/node-v72-linux-arm/better_sqlite3.node'
>  ]
>}

**Comment by zadam** [comment, marked as answer]
Hi, try running `npm rebuild` after `npm install`.

**Reply by Korozion** [reply]
Yes ! At first I thought it didn't work as i got multiple errors after 5 minutes of waiting, but despite the errors, the server is now running !

For anyone having the same output : 
```
npm rebuild :

npm ERR! code 1
npm ERR! path /home/osmc/trilium/node_modules/electron
npm ERR! command failed
npm ERR! command sh -c node install.js
npm ERR! HTTPError: Response code 404 (Not Found)
npm ERR!     at EventEmitter.<anonymous> (/home/osmc/trilium/node_modules/got/source/as-stream.js:35:24)
npm ERR!     at EventEmitter.emit (events.js:314:20)
npm ERR!     at module.exports (/home/osmc/trilium/node_modules/got/source/get-response.js:22:10)
npm ERR!     at ClientRequest.handleResponse (/home/osmc/trilium/node_modules/got/source/request-as-event-emitter.js:155:5)
npm ERR!     at Object.onceWrapper (events.js:421:26)
npm ERR!     at ClientRequest.emit (events.js:326:22)
npm ERR!     at ClientRequest.origin.emit (/home/osmc/trilium/node_modules/@szmarczak/http-timer/source/index.js:37:11)
npm ERR!     at HTTPParser.parserOnIncomingClient (_http_client.js:601:27)
npm ERR!     at HTTPParser.parserOnHeadersComplete (_http_common.js:122:17)
npm ERR!     at TLSSocket.socketOnData (_http_client.js:474:22)

npm ERR! A complete log of this run can be found in:
npm ERR!     /root/.npm/_logs/2021-09-01T19_28_57_452Z-debug.log
```
it may still work !

Thank you zadam

**Comment by MorganeLeCaignec** [comment]
I've been doing this on an RPI for multiple versions as well, and my method is to delete the `node_modules` folder and the `package-lock.json` file then do `npm install`. It takes a while but then `node src/www` works perfectly.

**Comment by Lewiscowles1986** [comment]
More-or less this worked. arm64 pi seemed to behave a bit crazy (file not found etc), due to libc, but after much tinkering and the notes from this page, I was able to get it to work. Thanks.
